internal/web/routes_actions_test.go

```diff
@@ -0,0 +1,63 @@
+package web
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+
+	"testing"
+
+	"github.com/amir20/dozzle/internal/docker"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_handler_containerActions(t *testing.T) {
+	mockedClient := new(MockedClient)
+	mockedClient.On("FindContainer", "123").Return(docker.Container{ID: "123"}, nil)
+	mockedClient.On("FindContainer", "456").Return(docker.Container{}, errors.New("container not found"))
+
+	mockedClient.On("ContainerActions", "start", docker.Container{ID: "123"}).Return(nil)
+	mockedClient.On("ContainerActions", "stop", docker.Container{ID: "123"}).Return(nil)
+	mockedClient.On("ContainerActions", "restart", docker.Container{ID: "123"}).Return(nil)
+	mockedClient.On("ContainerActions", "something-else", docker.Container{ID: "123"}).Return(errors.New("unknown action"))
+
+	mockedClient.On("ContainerActions", "start", mock.Anything).Return(errors.New("container not found"))
+
+	handler := createHandler(mockedClient, nil, Config{Base: "/", EnableActions: true, Authorization: Authorization{Provider: NONE}})
+	req, err := http.NewRequest("POST", "/api/actions/start/localhost/123", nil)
+	require.NoError(t, err, "Request should not return an error.")
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, req)
+	assert.Equal(t, rr.Code, 200)
+
+	req, err = http.NewRequest("POST", "/api/actions/stop/localhost/123", nil)
```

Comment:
Its generally bad practice to have multiple assertions in one test. It would be better to break up these tests where there is only one assert per test. This is also important because if one assert breaks, then the whole test will fail which is not great.